audio/out/buffer.c

```diff
@@ -62,6 +62,11 @@ struct buffer_state {
     bool paused;                // logically paused
 
     int64_t end_time_ns;        // absolute output time of last played sample
+    int64_t queued_time_ns;     // duration of samples that have been queued to
+                                // the device but have not been played.
+                                // This field is only set in ao_set_paused(),
+                                // and is considered as a temporary solution;
```

Comment:
did you mean "is only valid temporarily"?

Reply:
I mean, this field may be removed at any time in the future, if some better method comes up.

Reply:
Or maybe we can use a union for `end_time_ns` and `queued_time_ns` giving they can not be valid at the same time?

Reply:
An union would be the kind of too clever that I would like to avoid.

Reply:
Ok